src/clj/swarmpit/api.clj

```diff
@@ -60,6 +57,26 @@
   (->> (cmo/->password password)
        (cc/change-password user)))
 
+(def password-check hashers/check)
+
+(defn password-check-upgrade
+  [password hash upgrade-f]
+  (try
+    (password-check password hash)
+    (catch Exception e
+      (if (= hash (digest "sha-256" password))
+        (do (upgrade-f) true)
+        false))))
+
+(defn user-by-credentials
+  [credentials]
+  (let [{:keys [username password]} credentials
```

Comment:
i have to use let for `user` anyway, so it doesn't really matter that much, but a nice tip